src/mlpack/tests/transposed_convolutional_network_test.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file transposed_convolutional_network_test.cpp
+ * @author Shikhar Jaiswal
+ *
+ * Tests the transposed convolutional neural network.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+
+#include <mlpack/core/optimizers/rmsprop/rmsprop.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+
+BOOST_AUTO_TEST_SUITE(TransposedConvolutionalNetworkTest);
+
+/**
+ * Train the vanilla network on a larger dataset.
+ */
+BOOST_AUTO_TEST_CASE(VanillaNetworkTest)
```

Comment:
What do you think about adding a test that compares against some precomputed results? Just to test the ``TransposedConvolutional`` layer, also a gradient check test might be a good idea to add.

Reply:
@zoq Could you provide an example from the code where something like that is done? That would make it easier for me to implement the test and debug the current implementation, then I can go ahead and correct the `transposed_network_test.cpp` (which is currently just a copy of the `convolutional_network_test.cpp`).

Reply:
gradient check for the linear layer:

https://github.com/mlpack/mlpack/blob/e37ceb380c4cfb7559916b795b66231f6e2f90f9/src/mlpack/tests/ann_layer_test.cpp#L540-L578

Convolution test against a precomputed result:

https://github.com/mlpack/mlpack/blob/e37ceb380c4cfb7559916b795b66231f6e2f90f9/src/mlpack/tests/convolution_test.cpp#L157-L189